### PR TITLE
Add sample json search document

### PR DESCRIPTION
**Pull request by zhongh**
- mapping file in /mappings
- has 2 bulk data files, one is _pretty and one is _compact, just in case
- the current bulk file contains only one observation entry with all the properties, if this works I can copy-paste more entries in it.

### CODE REVIEW

**Comment by zednis** [comment]
Could you move the example JSON file to a /example directory?

edit: and the example mineral.csv.  Keep the mapping file in ingest/mappings.

File: ingest/example/bulk_pretty [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      },
+      "mindat":"http://www.mindat.org/loc-599.html"
+    }
+  ],
+  "description":"Named after the Abenaki, the indigenous people who inhabited the area around Mont Saint-Hilaire in southern Québec, prior to the arrival of European settlers.",
+  "mindat":"http://www.mindat.org/min-2.html",
+  "mineral":{  
+    "name":"Abenakiite-(Ce)",
+    "rruffchemistry":"Na_26_Ce^3+^_6_(SiO_3_)_6_(PO_4_)_6_(CO_3_)_6_(S^4+^O_2_)O",
+    "elements":"Na Ce Si O P C S"
```
Comment (zednis) [inline ingest/example/bulk_pretty:26]: Should these elements be separate values in an array?

@xgmachina thoughts?

File: ingest/example/bulk_pretty [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      "coordinates":{  
+        "lat":"45.56277778",
+        "lon":"-73.14166667"
+      },
+      "mindat":"http://www.mindat.org/loc-599.html"
+    }
+  ],
+  "description":"Named after the Abenaki, the indigenous people who inhabited the area around Mont Saint-Hilaire in southern Québec, prior to the arrival of European settlers.",
+  "mindat":"http://www.mindat.org/min-2.html",
+  "mineral":{
```
Comment (zednis) [inline ingest/example/bulk_pretty:22]: Should we have max age and min age for the observed mineral specimen?  (see #4, and ask @xgmachina about it)

File: ingest/example/bulk_pretty [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        "lon":"-73.14166667"
+      },
+      "mindat":"http://www.mindat.org/loc-599.html"
+    }
+  ],
+  "description":"Named after the Abenaki, the indigenous people who inhabited the area around Mont Saint-Hilaire in southern Québec, prior to the arrival of European settlers.",
+  "mindat":"http://www.mindat.org/min-2.html",
+  "mineral":{  
+    "name":"Abenakiite-(Ce)",
+    "rruffchemistry":"Na_26_Ce^3+^_6_(SiO_3_)_6_(PO_4_)_6_(CO_3_)_6_(S^4+^O_2_)O",
```
Comment (zednis) [inline ingest/example/bulk_pretty]: Is there a standard syntax for encoding this chemistry in text?

I see `Na26Ce6(Si6O18)(PO4)6(CO3)6(SO3)` at the the mandate link.

File: ingest/example/bulk_pretty [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "locality":[  
+    {  
+      "country":"Canada",
+      "region":"Québec, Canada",
+      "mine":"Poudrette quarry (Demix quarry; Uni-Mix quarry; Desourdy quarry; Carrière Mont Saint-Hilaire), Mont Saint-Hilaire, La Vallée-du-Richelieu RCM, Montérégie, Québec, Canada",
+      "coordinates":{  
+        "lat":"45.56277778",
+        "lon":"-73.14166667"
+      },
+      "mindat":"http://www.mindat.org/loc-599.html"
```
Comment (zednis) [inline ingest/example/bulk_pretty:17]: Why did you move the property mindat into the locality object?

edit: I see this is a mindat URL for the locality.  Let's call the property something other than mindat then; "locality_url" or simply "url".

**Comment by zednis** [comment]
Would it make sense to add properties for the following in the mineral object?

```
System:Hexagonal
Colour:Pale brown
Lustre:Vitreous
Hardness:4 - 5
```

**Comment by zednis** [comment]
We may want to separate the observed mineral specimen, which can have an estimated age, from the description of the mineral species.

@xgmachina what do you think?

**Comment by xgmachina** [comment]
I looked at some mindata records through the rruff/ima page. They show that the ages are with observations (instances of mineral species), not the mineral species. Or in other words, a same type of mineral may be formed at different periods in Earth's history.

**Comment by zednis** [comment]
Agreed, that is what I was suggesting.  In this case the mineral specimen is the instance of mineral observed at the locality.  So, separating the specimen (instance at locality) from the species.

**Comment by zednis** [comment]
Perhaps we should look at O&M and the Sampling Features ontology for general guidance?  Simon Cox wrote a paper on it recently that was in the Semantic Web Journal.

**Comment by zednis** [comment]
I'll put a link to Simon's paper in the slack channel

File: ingest/example/bulk_pretty (hunk 3 above)
Comment (zhongh) [inline ingest/example/bulk_pretty]: So the format I adopted ("Na_26_Ce^3+^_6_....) is one of the standard syntaces, where _x_ means subscription and ^x^ means superscription. 
Another syntax is using html format, where <sub>x</sub> means subscription and <sup>x</sup> means superscription.
These two syntaces are obviously interchangeable. Maybe we can use the html style since we can easily display it in the browser.

**Comment by zhongh** [comment]
The browser based on handlebars is working and I have got myself through how to use it
image
Made some little changes to the browser and the sample data, such as the html-style syntax of chemicals. But need to discuss with Marshall on records/locality/ages/mineral species/etc before we move-on with the browser.

File: html/browser.html [hunk 5]
```diff
@@ -4,7 +4,7 @@
     <meta charset="UTF-8">
     <title>DTDI Mineral Observation Map</title>
 
-    <script src="//cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.1/handlebars.min.js"></script>
+    <script src="handlebars.min.js"></script>
```
Comment (zednis) [inline html/browser.html]: 1) Why remove the handlebars CDN reference?
2) Did you add handlebar.min.js to the repo?

File: html/browser.html [hunk 6]
```diff
@@ -20,18 +20,43 @@
...
-                    {{!-- link to MINDAT profile (badge?) --}}
-                </div>
+                {{#if mineral.name}}<div><strong>Mineral Name:</strong> <a href="{{mindat}}" target="_blank"> {{mineral.name}}</a></div>
+                {{else}} <div><strong>Mineral Name:</strong> N/A</div>
+                {{/if}}
+
+                {{#if locality}}
+                <div> <strong>Locality (Need discussion of observation records, ages, etc with Marshall) </strong></div>
+                {{#expand locality}}
+                    <div>&nbsp&nbsp <strong>Mine:</strong> <a href="{{mindat}}" target="_blank"> {{mine}}</a></div>
```
Comment (zednis) [inline html/browser.html]: Let's remove usage of &nbsp and use CSS styles for layout.

File: ingest/example/bulk_pretty (hunk 2 above)
Comment (zednis) [inline ingest/example/bulk_pretty:22]: @zhonghao1014 bump

File: ingest/example/bulk_pretty [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    }
+  ],
+  "description":"Named after the Abenaki, the indigenous people who inhabited the area around Mont Saint-Hilaire in southern Québec, prior to the arrival of European settlers.",
+  "mindat":"http://www.mindat.org/min-2.html",
+  "mineral":{  
+    "name":"Abenakiite-(Ce)",
+    "rruffChemistryConcise":"Na_26_Ce^3+^_6_(SiO_3_)_6_(PO_4_)_6_(CO_3_)_6_(S^4+^O_2_)O",
+    "rruffChemistryHTML":"Na<sub>26</sub>Ce<sup>3+</sup><sub>6</sub>(SiO<sub>3</sub>)<sub>6</sub>(PO<sub>4</sub>)<sub>6</sub>(CO<sub>3</sub>)<sub>6</sub>(S<sup>4+</sup>O<sub>2</sub>)O",
+    "elements":"Na Ce Si O P C S"
```
Comment (zednis) [inline ingest/example/bulk_pretty:26]: I think these elements should be an array instead of whitespace delimited string.

`"elements": ["Na", "Ce", "Si", "O", "P", "C", "S"]`

What do you think @xgmachina ?

**Comment by zhongh** [comment]
@zednis Agree with all suggestions above. And plus some small details discussed with Marshall today I am working on getting a modified data structure with an update of the faceted browser on over the weekend so that next week we can work on the map and the ingest script together

File: ingest/example/bulk_pretty (hunk 7 above)
Comment (zhongh) [inline ingest/example/bulk_pretty:26]: I think an array is clearly the way to go.

File: html/browser.html (hunk 6 above)
Comment (zhongh) [inline html/browser.html]: :+1:

File: html/browser.html (hunk 5 above)
Comment (zhongh) [inline html/browser.html]: For some reason the does not work locally on my machine, so I download the JS script and put it in the same folder to use it.

**Comment by zhonghao1014** [comment]
This browser is working with the new data structure. More work needs to be done with the browser, but it should be on the right track and right data structure.
image

**Comment by zhongh** [comment]
This is a peek to the data:

{  
  "mineral":{  
    "mineral-name-plain":"Abelsonite",
    "chemistry-elements-list":[  
      "Ni",
      "C",
      "H",
      "N"
    ],
    "rruff-chemistry-html":"Ni<sup>2+</sup>C<sub>31</sub>H<sub>32</sub>N<sub>4</sub>",
    "mineral-mindat-url":"http://www.mindat.org/min-1.html",
    "year-first-published":"1975",
    "country-of-type-locality":"USA",
    "ima-number":"IMA1975-013",
    "fleischers-groupname":"\u00a0",
    "ima-chesmistry-html":"NiC<sub>31</sub>H<sub>32</sub>N<sub>4</sub>",
    "rruff-chemistry-concise":"Ni^2+^C_31_H_32_N_4_",
    "ima-chesmistry-concise":"NiC_31_H_32_N_4_"
  },
  "locality":{  
    "country":"USA",
    "dated-locality-mindat-id":"180470",
    "max-age":"330",
    "lon-deg":"-111",
    "decimal-degree":"39.653611111111,-111.59833333333",
    "dated-locality-label":"Blackbird Mine (Pioneer Mine),Big Indian District (Big Indian Wash - Lisbon Valley Area),San Juan Co., Utah, USA",
    "lon-sec":"-54",
    "min-age":"13",
    "lon-min":"-35",
    "locality-mindat-url":"http://www.mindat.org/loc-14378.html",
    "locality-mindat-id":"14378",
    "region":"Utah, USA",
    "lat-min":"39",
    "locality-label":"Utah, USA",
    "lat-sec":"13",
    "dated-locality-mindat-url":"http://www.mindat.org/loc-180470.html",
    "lat-deg":"39",
    "lon":-111.59833333333333,
    "lat":39.65361111111111
  }
}

**Comment by zhongh** [comment]
Mapping file updated and facets are displayed correctly:
image

**Comment by zednis** [comment]
resolves #8
